tests/performance_estimation/test_cbpe.py

```diff
@@ -231,14 +275,50 @@ def test_cbpe_for_regression_metadata_raises_not_implemented_error():  # noqa: D
 
 
 def test_cbpe_for_binary_classification_does_not_fail_when_fitting_with_subset_of_reference_data(  # noqa: D103
-    metadata, data
+    binary_classification_metadata, binary_classification_data
 ):
-    reference = data[0].loc[40000:, :]
-    estimator = CBPE(model_metadata=metadata, metrics=['f1'])
+    reference = binary_classification_data[0].loc[40000:, :]
+    estimator = CBPE(model_metadata=binary_classification_metadata, metrics=['f1'])
     try:
         estimator.fit(reference_data=reference)
     except KeyError:
         pytest.fail(
             'fitting on subset resulted in KeyError => misaligned indices between data and stratified shuffle'
             'split results.'
         )
+
+
+@pytest.mark.skip(reason="This is a test util function to factor code")
+def reduce_confidence_bounds(monkeypatch, ref_df, ana_df, metadata):
```

Comment:
@nnansters I would like feedback about that.

I have created the reduce_confidence_bounds function to avoid duplicated code in test_cbpe_for_binary_classification_does_not_output_confidence_bounds_outside_appropriate_interval and test_cbpe_for_multiclass_classification_does_not_output_confidence_bounds_outside_appropriate_interval since they are basically doing the same thing on different inputs. As reduce_confidence_bounds should not be called as a unit test I added the skip decorator with a comment. I am not sure about the results so if you have a better idea I am all ears 🙂

Reply:
Hey @rfrenoy ,

good instinct to turn this into a function (something I should do more often in test code). Two remarks:

1. You don't need the `@pytest.skip` decorator. Only functions whose name starts with `test_` are run as tests. Did a quick experiment that confirms this, so you can drop the decorator :+1: 
2. The `reduce_confidence_bounds` function now does more than what the title is describing: it modifies the confidence bounds, but it also performs the estimation AND checks the assertions. I like to structure my tests according to the `AAA-principle`: arrange - act - assert. The _arrange_ part creates the instances you need for the test and sets their states. The _act_ part is running the functionality you'd like to test. The _assert_ part is comparing the state of your instances to the expected state. Your utility function is now eating up the most important things of the test: running the `estimate` (the _act_) and checking the results (the _assert_). I'd prefer having a bit of duplication but fully grasping what my test is doing in a single look. So I'd just remove those few lines from the helper method and put them into the actual test function. 


Other than that: excellent commit, thanks for expanding the tests to multiclass as well!

Reply:
Thanks! I'll make the changes

Reply:
@nnansters it should be ok now 🙂

Reply:
Yes, very clean, love it!